metaflow/runtime.py

```diff
@@ -160,13 +163,27 @@ def __init__(
             for deco in step.decorators:
                 deco.runtime_init(flow, graph, package, self._run_id)
 
+    def _update_rerun_steps(self):
+        """
+        Any steps following steps to be rerun should also be included as
+        "rerun" steps. This is to ensure that the flow is executed correctly.
```

Comment:
maybe i am not following the comment here - but `rerun` steps are the ones that need to be executed during `resume`? if that is the case, why is this update needed?

Reply:
for a linear graph a -> b -> c -> d (and b, c succeeded) , when user `resume b`,  we want b and the successor of b (which is c) to rerun as well. Here this `update` function will include all the descendants.

In old resume, I think after this step-to-rerun, it goes back to normal execution mode. 

In the new resume, we need to avoid copying the successful successor tasks (c here) as well, and hence update this "steps-to-rerun" to rerun everything after b.

Reply:
I would recommend not creating a function would just lifting this logic alongside L:119 so that the code reads a bit easy. Not a whole lot of complexity is embedded in this method.

Reply:
done